runelite-client/src/main/java/net/runelite/client/plugins/inventoryviewer/InventoryViewerOverlay.java

```diff
@@ -78,6 +101,31 @@ public Dimension render(Graphics2D graphics)
 
 		final Item[] items = itemContainer.getItems();
 
+		if (config.groupItems())
+		{
+			// Store in a LinkedHashMap to preserve inventory ordering
+			final Map<Integer, Integer> totals = Arrays.stream(items)
+				.filter(p -> p.getId() != -1)
+				.collect(groupingBy(Item::getId, LinkedHashMap::new, summingInt(Item::getQuantity)));
+
+			for (Map.Entry<Integer, Integer> cursor : totals.entrySet())
+			{
+				final BufferedImage image = itemManager.getImage(cursor.getKey(), cursor.getValue(), true);
+				if (image != null)
+				{
+					panelComponent.getChildren().add(new ImageComponent(image));
+				}
+			}
+
+			// Add a placeholder if the inventory is empty, so the overlay can still be easily seen
+			if (totals.entrySet().size() == 0)
+			{
+				panelComponent.getChildren().add(inventoryIconSprite);
```

Comment:
I feel this icon should be somewhat transparent, similar to a bank placeholder, so it is as clear as possible that there are no items present.